tests/subsys/llext/hello_world/testcase.yaml

```diff
@@ -3,20 +3,28 @@ common:
   arch_allow:
     - arm
     - xtensa
-  filter: not CONFIG_MPU and not CONFIG_MMU and not CONFIG_SOC_SERIES_S32ZE_R52
   platform_exclude:
     - nuvoton_pfm_m487 # See #63167
 tests:
   llext.simple.readonly:
     arch_exclude: xtensa # for now
+    filter: not CONFIG_MPU and not CONFIG_MMU and not CONFIG_SOC_SERIES_S32ZE_R52
     extra_configs:
       - arch:arm:CONFIG_ARM_MPU=n
       - CONFIG_LLEXT_STORAGE_WRITABLE=n
+  llext.simple.readonly_mpu:
+    arch_exclude: xtensa # for now
+    filter: CONFIG_CPU_HAS_MPU and not CONFIG_LOG_PRINTK
```

Comment:
What is the issue with CONFIG_LOG_PRINTK? Maybe some note on commit message?

Reply:
LOG_PRINTK requires !USERSPACE and with LOG_PRINTK being forced on some builds the filter was needed to avoid CI errors